### DISCUSSION TITLE
Drag gets stuck at clamp limits

### DISCUSSION THREAD

**Discussion by carmandale** [pmndrs/use-gesture#274] [Q&A]
I want to be able to drag with your mouse, to move the scene left and right with limits.

Here is a code sandbox to demonstrate https://codesandbox.io/s/usedrag-gets-stuck-5we3x

When you get to the limits of the left and right movement, it gets "stuck" it seems that maybe it isn't really stuck because if you keep dragging in the opposite direction it will eventually start moving again - but makes for a user experience that feels broken.

I think that I may be using the clamp incorrectly.

thanks for any help,
Dale

**Comment by dbismut** [comment]
Hi Dale, didn't look at the sandbox in details (thanks for it anyway, a sandbox always help!!). Did you try using the `bounds` option from useDrag? 

https://use-gesture.netlify.app/docs/options/#bounds

Note that you don't need to use `{ pointerEvents: true }` anymore, react-use-gesture now uses pointer events by default.

**Reply by carmandale** [reply]
@dbismut ,
I think that fixes it. I did a code sandbox here https://codesandbox.io/s/usedrag-gets-stuck-bounds-vyfpo?file=/src/App.js

and will try it on my actual project and see if that does it.

thanks,
Dale
